src/Databases/DuckDB/fetchDuckDBTableStructure.cpp

```diff
@@ -0,0 +1,148 @@
+#include <Databases/DuckDB/fetchDuckDBTableStructure.h>
+
+#if USE_DUCKDB
+
+#include <Common/quoteString.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeDate.h>
+#include <DataTypes/DataTypeDate32.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <DataTypes/DataTypeFactory.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesDecimal.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/DataTypeUUID.h>
+#include <Poco/String.h>
+
+#include <string_view>
+
+
+using namespace std::literals;
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int DUCKDB_ENGINE_ERROR;
+    extern const int BAD_ARGUMENTS;
+}
+
+static DataTypePtr convertDuckDBDataType(String type)
```

Comment:
After that, we can simply use `DataTypeFactory` here and in every similar place.